src/tracer.cc

```diff
@@ -300,7 +300,9 @@ TraceManager::GetTraceStartOptions(
 {
   TraceManager::TraceStartOptions start_options;
   GetTraceSetting(model_name, start_options.trace_setting);
-  if (start_options.trace_setting->mode_ == TRACE_MODE_OPENTELEMETRY) {
+  if (!start_options.trace_setting->level_ ==
+          TRITONSERVER_TRACE_LEVEL_DISABLED &&
+      start_options.trace_setting->mode_ == TRACE_MODE_OPENTELEMETRY) {
```

Comment:
This fixes memory leak when `mode` is opentelemetry, but trace level is `OFF`, i.e. tracing disabled. Not happening in the default scenario, only when user specifies `mode=opentelemetry`, but doesn't specify `level=TIMESTAMPS`

Reply:
Can be follow-up if complicated, but shouldn't the code not get this far if `--trace-level OFF` (tracing disabled)?

I would expect something earlier in the call chain looking like:
```cpp
// ex: StartTrace() - don't know the entrypoint
Status SomeTraceEntrypointFunction() {
  if (level == TRITONSERVER_TRACE_LEVEL_DISABLED) {
    return Status(INVALID_ARG, "Tracing is disabled");
    // or depending on how code works
    return Status::Success; // do nothing when tracing disabled
  }
}
```

Reply:
The problem is `GetTraceStartOptions` is called from http/grpc code and if `mode` was specifically chosen as `opentelemetry` , we check for OTel context in headers, if those exist, we do a force sample. 

Thus, added check will force sample (based on the context) trace, only when `level` != `OFF`.

Reply:
I see, thanks for explaining 🙏